p2p/ipld/write.go

```diff
@@ -52,20 +52,26 @@ func PutBlock(
 	if err != nil {
 		return fmt.Errorf("failure to recompute the extended data square: %w", err)
 	}
+
 	// get row and col roots to be provided
 	// this also triggers adding data to DAG
-	prov := newProvider(ctx, croute, int32(squareSize*4), logger.With("height", block.Height))
-	for _, root := range eds.RowRoots() {
-		prov.Provide(plugin.MustCidFromNamespacedSha256(root))
-	}
-	for _, root := range eds.ColumnRoots() {
-		prov.Provide(plugin.MustCidFromNamespacedSha256(root))
-	}
+	rowRoots := eds.RowRoots()
+	colRoots := eds.ColumnRoots()
+
 	// commit the batch to ipfs
 	err = batchAdder.Commit()
 	if err != nil {
 		return err
 	}
+
+	prov := newProvider(ctx, croute, int32(squareSize*4), logger.With("height", block.Height))
+	for _, root := range rowRoots {
+		prov.Provide(plugin.MustCidFromNamespacedSha256(root))
+	}
+	for _, root := range colRoots {
+		prov.Provide(plugin.MustCidFromNamespacedSha256(root))
+	}
+
```

Comment:
Providing does not require nodes to be committed and this could potentially increase PutBlock execution time for multicore systems, as before starting providing we will wait while all the data is written. We should start providing as early as possible